database/sql/V048__create_settlement_rewards.sql

```diff
@@ -12,20 +14,20 @@ CREATE TABLE settlement_observations (
   -- table, read from the transaction receipt
   block_number bigint NOT NULL,
   log_index bigint NOT NULL,
-  gas_used numeric(78,0) NOT NULL,
-  effective_gas_price numeric(78,0) NOT NULL,
+  gas_used numeric(78,0),
+  effective_gas_price numeric(78,0),
   -- the surplus observed from the transaction call data,
   -- and converted to ETH with the auction external prices.
-  surplus numeric(78,0) NOT NULL,
-  fee numeric(78,0) NOT NULL,
+  surplus numeric(78,0),
+  fee numeric(78,0),
 
   PRIMARY KEY (block_number, log_index)
 );
 
 CREATE TABLE auction_prices (
-  auction_id bigint NOT NULL,
-  token bytea NOT NULL,
-  price numeric(78,0) NOT NULL
+  auction_id bigint NOT NULL PRIMARY KEY,
+  tokens bytea[] NOT NULL,
+  prices numeric(78,0)[] NOT NULL
 );
 
 CREATE TABLE auction_participants (
```

Comment:
I missed the PR that added this, but I also think that we should have 1 row per participant and not use arrays.

At least this is my simple understanding of how this "should" be (and allow you to query things like `SELECT COUNT(*) from auction_participants where participant = '\x...'` for querying the number of times an address participated in an auction).

Reply:
I would be very much in favour of not using arrays. They are hard to parse and also make transposition a two-step process!

Reply:
Fixed.